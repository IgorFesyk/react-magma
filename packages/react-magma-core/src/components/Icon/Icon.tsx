import * as React from 'react';

export class IconCore extends React.Component<IconCoreProps> {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.handleClick && this.props.handleClick();
  }

  render() {
    return this.props.children({
      ...this.props,
      handleClick: this.handleClick
    });
  }
}

export interface IconCoreProps {
  children: (props) => React.ReactNode;
  handleClick?: () => void;
}
