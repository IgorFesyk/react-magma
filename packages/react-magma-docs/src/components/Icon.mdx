---
name: Icon
menu: Components
---

import { Playground } from 'docz';
import { IconContainer } from '../docs-components/IconContainer';
import { IconListContainer } from '../docs-components/IconListContainer';
import { Icon, ICONS } from 'react-magma-dom';

# Icons

## Basic Usage

<Playground>
  <p>
    <IconContainer>
      <Icon id="infoBasic" title="Info Available" type="info" />
    </IconContainer>
  </p>
</Playground>

## Sizes

<Playground>
  <p>
    <IconContainer>
      <Icon id="smallSize" size={20} title="Small Size" type="angle-down" />
    </IconContainer>
    <IconContainer>
      <Icon id="largeSize" size={50} title="Large Size" type="angle-down" />
    </IconContainer>
  </p>
</Playground>

## Color

<Playground>
  <p>
    <IconContainer>
      <Icon id="noColor" title="No Color" type="angle-down" />
    </IconContainer>
    <IconContainer>
      <Icon id="someColor" color="green" title="Some Color" type="angle-down" />
    </IconContainer>
  </p>
</Playground>

## Available Icons

<Playground>
  {
    <IconListContainer>{Object.keys(ICONS).map((key) => {
      const id = key + 'Available';
      const title = key + ' Title';

      return <IconContainer border>
        <Icon id={id} title={title} type={key} /><br />
        {key}
      </IconContainer>
    })}</IconListContainer>
  }
</Playground>
