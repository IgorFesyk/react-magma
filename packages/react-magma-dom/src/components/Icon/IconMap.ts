import { AngleDownIcon } from './types/angle-down';
import { FolderOpenIcon } from './types/folder-open';
import { InfoIcon } from './types/info';

export const IconMap = {
  'angle-down': AngleDownIcon,
  'folder-open': FolderOpenIcon,
  info: InfoIcon
};
