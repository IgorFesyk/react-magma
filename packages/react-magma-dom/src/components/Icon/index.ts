import { AddPersonIcon } from './types/AddPersonIcon';
import { AlertIcon } from './types/AlertIcon';
import { AndroidIcon } from './types/AndroidIcon';
import { AngleDownIcon } from './types/AngleDownIcon';
import { AngleDownDoubleIcon } from './types/AngleDownDoubleIcon';
import { AngleLeftIcon } from './types/AngleLeftIcon';
import { AngleLeftDoubleIcon } from './types/AngleLeftDoubleIcon';
import { AngleRightIcon } from './types/AngleRightIcon';
import { AngleRightDoubleIcon } from './types/AngleRightDoubleIcon';
import { AngleUpIcon } from './types/AngleUpIcon';
import { AngleUpDoubleIcon } from './types/AngleUpDoubleIcon';
import { AppleIcon } from './types/AppleIcon';
import { ArrowDown2Icon } from './types/ArrowDown2Icon';
import { ArrowDown3Icon } from './types/ArrowDown3Icon';
import { ArrowLeft2Icon } from './types/ArrowLeft2Icon';
import { ArrowLeft3Icon } from './types/ArrowLeft3Icon';
import { ArrowRight2Icon } from './types/ArrowRight2Icon';
import { ArrowRight3Icon } from './types/ArrowRight3Icon';
import { ArrowUp2Icon } from './types/ArrowUp2Icon';
import { ArrowUp3Icon } from './types/ArrowUp3Icon';
import { ArticleIcon } from './types/ArticleIcon';
import { AssignIcon } from './types/AssignIcon';
import { AsteriskIcon } from './types/AsteriskIcon';
import { AttachmentIcon } from './types/AttachmentIcon';
import { BackwardIcon } from './types/BackwardIcon';
import { Backward2Icon } from './types/Backward2Icon';
import { BankIcon } from './types/BankIcon';
import { BellIcon } from './types/BellIcon';
import { BitcoinIcon } from './types/BitcoinIcon';
import { BlockedIcon } from './types/BlockedIcon';
import { Book2Icon } from './types/Book2Icon';
import { BookmarkIcon } from './types/BookmarkIcon';
import { BubbleIcon } from './types/BubbleIcon';
import { Bug2Icon } from './types/Bug2Icon';
import { CalendarIcon } from './types/CalendarIcon';
import { CancelCircleIcon } from './types/CancelCircleIcon';
import { CaretDownIcon } from './types/CaretDownIcon';
import { CaretLeftIcon } from './types/CaretLeftIcon';
import { CaretRightIcon } from './types/CaretRightIcon';
import { CaretUpIcon } from './types/CaretUpIcon';
import { CartIcon } from './types/CartIcon';
import { CcAmexIcon } from './types/CcAmexIcon';
import { CcDiscoverIcon } from './types/CcDiscoverIcon';
import { CcMastercardIcon } from './types/CcMastercardIcon';
import { CcPaypalIcon } from './types/CcPaypalIcon';
import { CcStripeIcon } from './types/CcStripeIcon';
import { CcVisaIcon } from './types/CcVisaIcon';
import { CheckIcon } from './types/CheckIcon';
import { CircleDownIcon } from './types/CircleDownIcon';
import { CircleLeftIcon } from './types/CircleLeftIcon';
import { CircleRightIcon } from './types/CircleRightIcon';
import { CircleUpIcon } from './types/CircleUpIcon';
import { ClockIcon } from './types/ClockIcon';
import { Cloud2Icon } from './types/Cloud2Icon';
import { CnyIcon } from './types/CnyIcon';
import { CodeIcon } from './types/CodeIcon';
import { Cog2Icon } from './types/Cog2Icon';
import { CommentsIcon } from './types/CommentsIcon';
import { CompassIcon } from './types/CompassIcon';
import { ConnectionIcon } from './types/ConnectionIcon';
import { CopyIcon } from './types/CopyIcon';
import { CountsTowardsGradeIcon } from './types/CountsTowardsGradeIcon';
import { CourseIcon } from './types/CourseIcon';
import { CreditNoCreditIcon } from './types/CreditNoCreditIcon';
import { CrossIcon } from './types/CrossIcon';
import { CubeIcon } from './types/CubeIcon';
import { DashboardIcon } from './types/DashboardIcon';
import { DollarIcon } from './types/DollarIcon';
import { DotCircleOIcon } from './types/DotCircleOIcon';
import { Download2Icon } from './types/Download2Icon';
import { DroppedIcon } from './types/DroppedIcon';
import { EllipsisHIcon } from './types/EllipsisHIcon';
import { EllipsisVIcon } from './types/EllipsisVIcon';
import { EnlargeIcon } from './types/EnlargeIcon';
import { Enlarge2Icon } from './types/Enlarge2Icon';
import { EntypoIcon } from './types/EntypoIcon';
import { EnvelopeOIcon } from './types/EnvelopeOIcon';
import { EnvelopeIcon } from './types/EnvelopeIcon';
import { ErrorBlackIcon } from './types/ErrorBlackIcon';
import { EurIcon } from './types/EurIcon';
import { ExamIcon } from './types/ExamIcon';
import { ExclamationIcon } from './types/ExclamationIcon';
import { ExternalLinkIcon } from './types/ExternalLinkIcon';
import { Extracredit2Icon } from './types/Extracredit2Icon';
import { EyeSlashIcon } from './types/EyeSlashIcon';
import { Eye2Icon } from './types/Eye2Icon';
import { FacebookIcon } from './types/FacebookIcon';
import { Feed5Icon } from './types/Feed5Icon';
import { FileArchiveOIcon } from './types/FileArchiveOIcon';
import { FileAudioOIcon } from './types/FileAudioOIcon';
import { FileCodeOIcon } from './types/FileCodeOIcon';
import { FileExcelOIcon } from './types/FileExcelOIcon';
import { FileImageOIcon } from './types/FileImageOIcon';
import { FileMovieOIcon } from './types/FileMovieOIcon';
import { FileOIcon } from './types/FileOIcon';
import { FilePdfOIcon } from './types/FilePdfOIcon';
import { FilePowerpointOIcon } from './types/FilePowerpointOIcon';
import { FileWordOIcon } from './types/FileWordOIcon';
import { Filter2Icon } from './types/Filter2Icon';
import { FirstIcon } from './types/FirstIcon';
import { FlagOIcon } from './types/FlagOIcon';
import { Flag3Icon } from './types/Flag3Icon';
import { FlashcardsIcon } from './types/FlashcardsIcon';
import { FolderOIcon } from './types/FolderOIcon';
import { FolderOpenOIcon } from './types/FolderOpenOIcon';
import { ForwardIcon } from './types/ForwardIcon';
import { FontSizeIcon } from './types/FontSizeIcon';
import { Forward2Icon } from './types/Forward2Icon';
import { Forward3Icon } from './types/Forward3Icon';
import { GbpIcon } from './types/GbpIcon';
import { GlassdoorIcon } from './types/GlassdoorIcon';
import { GoogleDriveIcon } from './types/GoogleDriveIcon';
import { GooglePlusIcon } from './types/GooglePlusIcon';
import { GoogleplayIcon } from './types/GoogleplayIcon';
import { GraduationCapIcon } from './types/GraduationCapIcon';
import { GroupIcon } from './types/GroupIcon';
import { HighlightIcon } from './types/HighlightIcon';
import { Home32Icon } from './types/Home32Icon';
import { ImageIcon } from './types/ImageIcon';
import { ImagesIcon } from './types/ImagesIcon';
import { InProgressIcon } from './types/InProgressIcon';
import { IndentDecreaseIcon } from './types/IndentDecreaseIcon';
import { IndentIncreaseIcon } from './types/IndentIncreaseIcon';
import { InfoIcon } from './types/InfoIcon';
import { Info2Icon } from './types/Info2Icon';
import { Info22Icon } from './types/Info22Icon';
import { InrIcon } from './types/InrIcon';
import { InstagramIcon } from './types/InstagramIcon';
import { InteractiveIcon } from './types/InteractiveIcon';
import { Key3Icon } from './types/Key3Icon';
import { KrwIcon } from './types/KrwIcon';
import { LastIcon } from './types/LastIcon';
import { LatePenaltyIcon } from './types/LatePenaltyIcon';
import { LifebuoyIcon } from './types/LifebuoyIcon';
import { LightningIcon } from './types/LightningIcon';
import { LinkIcon } from './types/LinkIcon';
import { LinkedinIcon } from './types/LinkedinIcon';
import { ListNumberedIcon } from './types/ListNumberedIcon';
import { ListIcon } from './types/ListIcon';
import { List2Icon } from './types/List2Icon';
import { List3Icon } from './types/List3Icon';
import { LockIcon } from './types/LockIcon';
import { ManualGradedIcon } from './types/ManualGradedIcon';
import { MapMarkerIcon } from './types/MapMarkerIcon';
import { MarketIcon } from './types/MarketIcon';
import { MenuSteppedIcon } from './types/MenuSteppedIcon';
import { MenuIcon } from './types/MenuIcon';
import { Menu2Icon } from './types/Menu2Icon';
import { Menu3Icon } from './types/Menu3Icon';
import { Menu4Icon } from './types/Menu4Icon';
import { MicrophoneSlashIcon } from './types/MicrophoneSlashIcon';
import { MicrophoneIcon } from './types/MicrophoneIcon';
import { MinusCircleIcon } from './types/MinusCircleIcon';
import { MinusIcon } from './types/MinusIcon';
import { MTAct1TimeIcon } from './types/MTAct1TimeIcon';
import { MTActLearnIcon } from './types/MTActLearnIcon';
import { MTActSerialIcon } from './types/MTActSerialIcon';
import { MtIcon } from './types/MtIcon';
import { Music2Icon } from './types/Music2Icon';
import { NextIcon } from './types/NextIcon';
import { Next2Icon } from './types/Next2Icon';
import { NonMindtapActivityIcon } from './types/NonMindtapActivityIcon';
import { NotificationIcon } from './types/NotificationIcon';
import { Notification2Icon } from './types/Notification2Icon';
import { PaletteIcon } from './types/PaletteIcon';
import { ParagraphCenterIcon } from './types/ParagraphCenterIcon';
import { ParagraphJustifyIcon } from './types/ParagraphJustifyIcon';
import { ParagraphLeftIcon } from './types/ParagraphLeftIcon';
import { ParagraphRightIcon } from './types/ParagraphRightIcon';
import { PauseIcon } from './types/PauseIcon';
import { Pause2Icon } from './types/Pause2Icon';
import { Pencil3Icon } from './types/Pencil3Icon';
import { PhoneHangUpIcon } from './types/PhoneHangUpIcon';
import { PhoneIcon } from './types/PhoneIcon';
import { PieChartIcon } from './types/PieChartIcon';
import { PinIcon } from './types/PinIcon';
import { PinterestIcon } from './types/PinterestIcon';
import { Play2Icon } from './types/Play2Icon';
import { Play3Icon } from './types/Play3Icon';
import { PlusIcon } from './types/PlusIcon';
import { PracticeIcon } from './types/PracticeIcon';
import { PreviousIcon } from './types/PreviousIcon';
import { Previous2Icon } from './types/Previous2Icon';
import { PrintIcon } from './types/PrintIcon';
import { ProfileIcon } from './types/ProfileIcon';
import { QuestionCircleOIcon } from './types/QuestionCircleOIcon';
import { QuestionCircleIcon } from './types/QuestionCircleIcon';
import { QuestionIcon } from './types/QuestionIcon';
import { Question2Icon } from './types/Question2Icon';
import { Reading2Icon } from './types/Reading2Icon';
import { RedoIcon } from './types/RedoIcon';
import { Redo2Icon } from './types/Redo2Icon';
import { RefreshIcon } from './types/RefreshIcon';
import { ReplyIcon } from './types/ReplyIcon';
import { RoubleIcon } from './types/RoubleIcon';
import { SearchMinusIcon } from './types/SearchMinusIcon';
import { SearchPlusIcon } from './types/SearchPlusIcon';
import { Search2Icon } from './types/Search2Icon';
import { ShrinkIcon } from './types/ShrinkIcon';
import { Shrink2Icon } from './types/Shrink2Icon';
import { SignInIcon } from './types/SignInIcon';
import { SnapchatIcon } from './types/SnapchatIcon';
import { Spinner2Icon } from './types/Spinner2Icon';
import { SpotifyIcon } from './types/SpotifyIcon';
import { StackIcon } from './types/StackIcon';
import { StarFullIcon } from './types/StarFullIcon';
import { StarHalfIcon } from './types/StarHalfIcon';
import { StarIcon } from './types/StarIcon';
import { StatsBarsIcon } from './types/StatsBarsIcon';
import { StatsBars2Icon } from './types/StatsBars2Icon';
import { StatsDotsIcon } from './types/StatsDotsIcon';
import { StopIcon } from './types/StopIcon';
import { Stop2Icon } from './types/Stop2Icon';
import { SwapHorizIcon } from './types/SwapHorizIcon';
import { TargetIcon } from './types/TargetIcon';
import { ThListIcon } from './types/ThListIcon';
import { ThIcon } from './types/ThIcon';
import { ThumbsODownIcon } from './types/ThumbsODownIcon';
import { ThumbsOUpIcon } from './types/ThumbsOUpIcon';
import { TimedIcon } from './types/TimedIcon';
import { TimerIcon } from './types/TimerIcon';
import { TrashOIcon } from './types/TrashOIcon';
import { TruckIcon } from './types/TruckIcon';
import { TuneIcon } from './types/TuneIcon';
import { TwitterIcon } from './types/TwitterIcon';
import { UnarchiveIcon } from './types/UnarchiveIcon';
import { UndoIcon } from './types/UndoIcon';
import { Undo2Icon } from './types/Undo2Icon';
import { UnlockIcon } from './types/UnlockIcon';
import { UnlockedIcon } from './types/UnlockedIcon';
import { Upload2Icon } from './types/Upload2Icon';
import { VideoCameraIcon } from './types/VideoCameraIcon';
import { VideoIcon } from './types/VideoIcon';
import { VolumeDownIcon } from './types/VolumeDownIcon';
import { VolumeOffIcon } from './types/VolumeOffIcon';
import { VolumeUpIcon } from './types/VolumeUpIcon';
import { WarningIcon } from './types/WarningIcon';
import { WidgetsIcon } from './types/WidgetsIcon';
import { Windows8Icon } from './types/Windows8Icon';
import { WlGotItIcon } from './types/WlGotItIcon';
import { WlLearnItIcon } from './types/WlLearnItIcon';
import { WlPracticeItIcon } from './types/WlPracticeItIcon';
import { WlReadyIcon } from './types/WlReadyIcon';
import { WlUseItIcon } from './types/WlUseItIcon';
import { WrenchIcon } from './types/WrenchIcon';
import { Wrench3Icon } from './types/Wrench3Icon';
import { YoutubeIcon } from './types/YoutubeIcon';
import { IconProps } from './utils';

interface IconsCollection {
  [key: string]: React.FunctionComponent<IconProps>;
}

export const ICONS: IconsCollection = {
  AddPersonIcon,
  AlertIcon,
  AndroidIcon,
  AngleDownIcon,
  AngleDownDoubleIcon,
  AngleLeftIcon,
  AngleLeftDoubleIcon,
  AngleRightIcon,
  AngleRightDoubleIcon,
  AngleUpIcon,
  AngleUpDoubleIcon,
  AppleIcon,
  ArrowDown2Icon,
  ArrowDown3Icon,
  ArrowLeft2Icon,
  ArrowLeft3Icon,
  ArrowRight2Icon,
  ArrowRight3Icon,
  ArrowUp2Icon,
  ArrowUp3Icon,
  ArticleIcon,
  AssignIcon,
  AsteriskIcon,
  AttachmentIcon,
  BackwardIcon,
  Backward2Icon,
  BankIcon,
  BellIcon,
  BitcoinIcon,
  BlockedIcon,
  Book2Icon,
  BookmarkIcon,
  BubbleIcon,
  Bug2Icon,
  CalendarIcon,
  CancelCircleIcon,
  CaretDownIcon,
  CaretLeftIcon,
  CaretRightIcon,
  CaretUpIcon,
  CartIcon,
  CcAmexIcon,
  CcDiscoverIcon,
  CcMastercardIcon,
  CcPaypalIcon,
  CcStripeIcon,
  CcVisaIcon,
  CheckIcon,
  CircleDownIcon,
  CircleLeftIcon,
  CircleRightIcon,
  CircleUpIcon,
  ClockIcon,
  Cloud2Icon,
  CnyIcon,
  CodeIcon,
  Cog2Icon,
  CommentsIcon,
  CompassIcon,
  ConnectionIcon,
  CopyIcon,
  CountsTowardsGradeIcon,
  CourseIcon,
  CreditNoCreditIcon,
  CrossIcon,
  CubeIcon,
  DashboardIcon,
  DollarIcon,
  DotCircleOIcon,
  Download2Icon,
  DroppedIcon,
  EllipsisHIcon,
  EllipsisVIcon,
  EnlargeIcon,
  Enlarge2Icon,
  EntypoIcon,
  EnvelopeOIcon,
  EnvelopeIcon,
  ErrorBlackIcon,
  EurIcon,
  ExamIcon,
  ExclamationIcon,
  ExternalLinkIcon,
  Extracredit2Icon,
  EyeSlashIcon,
  Eye2Icon,
  FacebookIcon,
  Feed5Icon,
  FileArchiveOIcon,
  FileAudioOIcon,
  FileCodeOIcon,
  FileExcelOIcon,
  FileImageOIcon,
  FileMovieOIcon,
  FileOIcon,
  FilePdfOIcon,
  FilePowerpointOIcon,
  FileWordOIcon,
  Filter2Icon,
  FirstIcon,
  FlagOIcon,
  Flag3Icon,
  FlashcardsIcon,
  FolderOIcon,
  FolderOpenOIcon,
  FontSizeIcon,
  ForwardIcon,
  Forward2Icon,
  Forward3Icon,
  GbpIcon,
  GlassdoorIcon,
  GoogleDriveIcon,
  GooglePlusIcon,
  GoogleplayIcon,
  GraduationCapIcon,
  GroupIcon,
  HighlightIcon,
  Home32Icon,
  ImageIcon,
  ImagesIcon,
  InProgressIcon,
  IndentDecreaseIcon,
  IndentIncreaseIcon,
  InfoIcon,
  Info2Icon,
  Info22Icon,
  InrIcon,
  InstagramIcon,
  InteractiveIcon,
  Key3Icon,
  KrwIcon,
  LastIcon,
  LatePenaltyIcon,
  LifebuoyIcon,
  LightningIcon,
  LinkIcon,
  LinkedinIcon,
  ListNumberedIcon,
  ListIcon,
  List2Icon,
  List3Icon,
  LockIcon,
  ManualGradedIcon,
  MapMarkerIcon,
  MarketIcon,
  MenuSteppedIcon,
  MenuIcon,
  Menu2Icon,
  Menu3Icon,
  Menu4Icon,
  MicrophoneSlashIcon,
  MicrophoneIcon,
  MinusCircleIcon,
  MinusIcon,
  MTAct1TimeIcon,
  MTActLearnIcon,
  MTActSerialIcon,
  MtIcon,
  Music2Icon,
  NextIcon,
  Next2Icon,
  NonMindtapActivityIcon,
  NotificationIcon,
  Notification2Icon,
  PaletteIcon,
  ParagraphCenterIcon,
  ParagraphJustifyIcon,
  ParagraphLeftIcon,
  ParagraphRightIcon,
  PauseIcon,
  Pause2Icon,
  Pencil3Icon,
  PhoneHangUpIcon,
  PhoneIcon,
  PieChartIcon,
  PinIcon,
  PinterestIcon,
  Play2Icon,
  Play3Icon,
  PlusIcon,
  PracticeIcon,
  PreviousIcon,
  Previous2Icon,
  PrintIcon,
  ProfileIcon,
  QuestionCircleOIcon,
  QuestionCircleIcon,
  QuestionIcon,
  Question2Icon,
  Reading2Icon,
  RedoIcon,
  Redo2Icon,
  RefreshIcon,
  ReplyIcon,
  RoubleIcon,
  SearchMinusIcon,
  SearchPlusIcon,
  Search2Icon,
  ShrinkIcon,
  Shrink2Icon,
  SignInIcon,
  SnapchatIcon,
  Spinner2Icon,
  SpotifyIcon,
  StackIcon,
  StarFullIcon,
  StarHalfIcon,
  StarIcon,
  StatsBarsIcon,
  StatsBars2Icon,
  StatsDotsIcon,
  StopIcon,
  Stop2Icon,
  SwapHorizIcon,
  TargetIcon,
  ThListIcon,
  ThIcon,
  ThumbsODownIcon,
  ThumbsOUpIcon,
  TimedIcon,
  TimerIcon,
  TrashOIcon,
  TruckIcon,
  TuneIcon,
  TwitterIcon,
  UnarchiveIcon,
  UndoIcon,
  Undo2Icon,
  UnlockIcon,
  UnlockedIcon,
  Upload2Icon,
  VideoCameraIcon,
  VideoIcon,
  VolumeDownIcon,
  VolumeOffIcon,
  VolumeUpIcon,
  WarningIcon,
  WidgetsIcon,
  Windows8Icon,
  WlGotItIcon,
  WlLearnItIcon,
  WlPracticeItIcon,
  WlReadyIcon,
  WlUseItIcon,
  WrenchIcon,
  Wrench3Icon,
  YoutubeIcon
};
