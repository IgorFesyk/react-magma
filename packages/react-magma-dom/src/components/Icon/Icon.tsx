import * as React from 'react';
import { IconMap } from './IconMap';
const styled = require('styled-components').default;

export interface IconProps {
  id: string;
  title: string;
  type: string;
  color?: string;
  size?: number;
}

function renderIcon(props) {
  const { id, title, type, color, size } = props;
  const IconType = IconMap[type];

  return IconType ? (
    <span className={type}>
      <IconType id={id} title={title} color={color} size={size} />
    </span>
  ) : null;
}

export const Icon: React.SFC<IconProps> = (props: IconProps): JSX.Element =>
  renderIcon(props);

export default Icon;
