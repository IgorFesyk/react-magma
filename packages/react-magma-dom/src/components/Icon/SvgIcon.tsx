import * as React from 'react';
import { useGenerateId } from '../utils';

const defaultSize = 24;

interface Path {
  d: string;
  transform?: string;
}

interface SvgIconProps extends React.HTMLAttributes<HTMLOrSVGElement> {
  id: string;
  testId?: string;
  title?: string;
  viewBox: string;
  paths: Path[];
  color?: string;
  size?: number;
}

function renderPaths(paths) {
  return paths.map(({ d, transform }, index) => (
    <path key={index} d={d} transform={transform} />
  ));
}

export const SvgIcon: React.FunctionComponent<SvgIconProps> = (
  props: SvgIconProps
) => {
  const { color, id: defaultId, size, title, testId, paths, ...other } = props;
  const id = useGenerateId(defaultId);

  return (
    <svg
      {...other}
      className="icon"
      height={size || defaultSize}
      width={size || defaultSize}
      fill={color || 'currentColor'}
      xmlns="http://www.w3.org/2000/svg"
      aria-labelledby={title ? id : null}
      data-testid={testId}
    >
      {title && <title id={id}>{title}</title>}
      {renderPaths(paths)}
    </svg>
  );
};
