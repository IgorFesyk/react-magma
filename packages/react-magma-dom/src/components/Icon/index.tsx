import * as React from 'react';
import { ICONS } from './types/icons';
import { SvgIcon } from './SvgIcon';

export interface IconProps {
  id?: string;
  title?: string;
  type: string;
  color?: string;
  size?: number;
}

function renderIcon(props) {
  const { id, title, type, color, size } = props;
  const iconType = ICONS[type];

  return iconType ? (
    <SvgIcon id={id} title={title} color={color} size={size} {...iconType} />
  ) : null;
}

export const Icon: React.FunctionComponent<IconProps> = (props: IconProps) =>
  renderIcon(props);
