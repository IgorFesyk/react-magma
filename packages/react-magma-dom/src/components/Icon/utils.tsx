import * as React from 'react';
import { SvgIcon } from './SvgIcon';

export interface IconProps {
  id?: string;
  title?: string;
  color?: string;
  size?: number;
}

export function renderIcon(props: IconProps, iconType: any) {
  const { id, title, color, size } = props;

  return iconType ? (
    <SvgIcon id={id} title={title} color={color} size={size} {...iconType} />
  ) : null;
}
