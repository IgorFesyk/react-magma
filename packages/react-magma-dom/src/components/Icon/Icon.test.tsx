import * as React from 'react';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import { Icon, IconProps } from './Icon';
import { render, fireEvent, cleanup } from 'react-testing-library';

const BASE_ICON_PROPS: IconProps = {
  id: 'infoId',
  title: 'Info Title',
  type: 'info'
};

const renderIcon = (myProps = {}) => {
  const props = {
    ...BASE_ICON_PROPS,
    ...myProps
  };

  return render(<Icon {...props} />);
};

describe('Icon', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render a icon with the passed in props', () => {
    const { container } = renderIcon();
    const title = container.firstChild.firstChild.childNodes[0];
    const svg = container.firstChild.firstChild.childNodes[1];

    expect(title).toHaveAttribute('id', BASE_ICON_PROPS.id);
    expect(title).toHaveTextContent(BASE_ICON_PROPS.title);
    expect(svg).toHaveAttribute('d');
    expect(svg).toHaveAttribute('transform');
  });

  it('should not render a non-mapped icon', () => {
    const { container } = renderIcon({
      type: 'failure'
    });

    expect(container.firstChild).toBeNull();
  });

  it('should render with different color', () => {
    const color = 'red';
    const { container } = renderIcon({ color });
    const svg = container.firstChild.firstChild;

    expect(svg).toHaveAttribute('fill', color);
  });

  it('should render with different size', () => {
    const size = '32';
    const { container } = renderIcon({ size });
    const svg = container.firstChild.firstChild;

    expect(svg).toHaveAttribute('height', size);
    expect(svg).toHaveAttribute('width', size);
  });

  describe('Snapshot Tests', () => {
    it('should render an info icon', () => {
      renderForSnapshots('info');
    });

    it('should render an angle-down icon', () => {
      renderForSnapshots('angle-down');
    });

    it('should render an folder-open icon', () => {
      renderForSnapshots('folder-open');
    });
  });
});

function renderForSnapshots(type) {
  const { container } = renderIcon({ type });

  expect(container).toMatchSnapshot();
}
