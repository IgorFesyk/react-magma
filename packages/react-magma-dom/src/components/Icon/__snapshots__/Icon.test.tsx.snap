// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Snapshot Tests should render an angle-down icon 1`] = `
<div>
  <span
    class="angle-down"
  >
    <svg
      aria-labelledby="infoId"
      class="icon"
      fill="currentColor"
      height="24"
      viewBox="0 0 15.59 9.09"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="infoId"
      >
        angle-down icon
      </title>
      <path
        d="M16.79,10.75a.49.49,0,0,1-.15.36L9.36,18.39a.54.54,0,0,1-.36.16.5.5,0,0,1-.36-.16L1.36,11.11a.49.49,0,0,1,0-.72l.78-.78a.5.5,0,0,1,.36-.16.54.54,0,0,1,.36.16L9,15.75l6.14-6.14a.5.5,0,0,1,.36-.16.54.54,0,0,1,.36.16l.78.78A.49.49,0,0,1,16.79,10.75Z"
        transform="translate(-1.2 -9.45)"
      />
    </svg>
  </span>
</div>
`;

exports[`Icon Snapshot Tests should render an folder-open icon 1`] = `
<div>
  <span
    class="folder-open"
  >
    <svg
      aria-labelledby="infoId"
      class="icon"
      fill="currentColor"
      height="24"
      viewBox="0 0 29.83 22"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="infoId"
      >
        folder-open icon
      </title>
      <path
        d="M27.91,15.55c0-.37-.27-.55-.83-.55h-17a3,3,0,0,0-1.33.34,3,3,0,0,0-1.12.82L3,21.83a1.13,1.13,0,0,0-.28.62c0,.37.27.55.82.55h17a3.13,3.13,0,0,0,1.35-.34A3.17,3.17,0,0,0,23,21.83l4.59-5.67A1,1,0,0,0,27.91,15.55ZM10.08,13h12V10.5a1.43,1.43,0,0,0-.43-1.06A1.48,1.48,0,0,0,20.58,9h-9a1.46,1.46,0,0,1-1.06-.44,1.44,1.44,0,0,1-.44-1.06v-1a1.43,1.43,0,0,0-.43-1.06A1.48,1.48,0,0,0,8.58,5h-5a1.46,1.46,0,0,0-1.06.44A1.44,1.44,0,0,0,2.08,6.5V19.83l4-4.92A5.23,5.23,0,0,1,7.9,13.54,5,5,0,0,1,10.08,13Zm19.83,2.55a3,3,0,0,1-.72,1.87l-4.61,5.67a4.93,4.93,0,0,1-1.81,1.37,5.11,5.11,0,0,1-2.19.54h-17a3.38,3.38,0,0,1-2.47-1,3.4,3.4,0,0,1-1-2.47V6.5A3.4,3.4,0,0,1,1.11,4,3.36,3.36,0,0,1,3.58,3h5a3.36,3.36,0,0,1,2.47,1,3.36,3.36,0,0,1,1,2.47V7h8.5a3.52,3.52,0,0,1,3.5,3.5V13h3a3.23,3.23,0,0,1,1.55.38,2.45,2.45,0,0,1,1.05,1.11,2.52,2.52,0,0,1,.23,1.06Z"
        transform="translate(-0.08 -3)"
      />
    </svg>
  </span>
</div>
`;

exports[`Icon Snapshot Tests should render an info icon 1`] = `
<div>
  <span
    class="info"
  >
    <svg
      aria-labelledby="infoId"
      class="icon"
      fill="currentColor"
      height="24"
      viewBox="0 0 25.63 25.63"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="infoId"
      >
        info icon
      </title>
      <path
        d="M15.26,10.15V7.56H12.74v2.59Zm0,10.29v-7.7H12.74v7.7ZM14,1.18A12.82,12.82,0,1,1,1.18,14,12.79,12.79,0,0,1,14,1.18Z"
        transform="translate(-1.18 -1.18)"
      />
    </svg>
  </span>
</div>
`;
